kivy/__init__.py

```diff
@@ -348,16 +348,21 @@ def _patch_mod_deps_win(dep_mod, mod_name):
 
 if not environ.get('KIVY_DOC_INCLUDE'):
     # Configuration management
+    user_home_dir = expanduser('~')
+    kivy_home_dir = None
+
     if 'KIVY_HOME' in environ:
         kivy_home_dir = expanduser(environ['KIVY_HOME'])
-    else:
-        user_home_dir = expanduser('~')
+    elif platform in {'android', 'ios'}:
```

Comment:
The below would be slightly more efficient in prevent 2 checks for "ios" / "android".
```python

    if 'KIVY_HOME' in environ:
        kivy_home_dir = expanduser(environ['KIVY_HOME'])
    elif platform == 'android':
        user_home_dir = environ['ANDROID_APP_PATH']
    elif platform == 'ios':
        user_home_dir = join(user_home_dir, 'Documents')
    elif sys.prefix != sys.base_prefix:
        # Detection if venv being used with the framework
        user_home_dir = sys.path[0]
```

Reply:
Okay, that's done but it seems I messed up the PR..

Reply:
From what I can tell, it's something like:
```
git remote add upstream https://github.com/kivy/kivy
git fetch upstream
git checkout stop-effect-in-scroll_to
git merge upstream/main
git rebase upstream/main
git add .
git rebase --continue # or git commit if you used merge
git push origin stop-effect-in-scroll_to
```
Can't really verify that, but hope it works for you?